xbmc/video/windows/GUIWindowVideoBase.cpp

```diff
@@ -638,8 +638,12 @@ bool CGUIWindowVideoBase::OnSelect(int iItem)
       !StringUtils::StartsWith(path, "newplaylist://") &&
       !StringUtils::StartsWith(path, "newtag://") &&
       !StringUtils::StartsWith(path, "script://"))
-    return OnFileAction(iItem, CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_MYVIDEOS_SELECTACTION), "");
-
+  {
+      if (item->GetVideoInfoTag()->m_type == MediaTypeEpisode)                                                                                            
+          return OnFileAction(iItem, CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_MYVIDEOS_SELECTEPISODEACTION), "");  // Episodes
+      else
+      return OnFileAction(iItem, CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(CSettings::SETTING_MYVIDEOS_SELECTACTION), "");             // Movies and others
```

Comment:
Coding guidelines! This line is way to long and indentation is wrong. Use clang-format to get the formatting right.

Reply:
How about streamlining this code a bit:

```c++
int iSelectAction = item->GetVideoInfoTag()->m_type == MediaTypeEpisode ? CSettings::SETTING_MYVIDEOS_SELECTEPISODEACTION : CSettings::SETTING_MYVIDEOS_SELECTACTION;
return OnFileAction(iItem, CServiceBroker::GetSettingsComponent()->GetSettings()->GetInt(iSelectAction), "");
```